source/develop/release-management/features/network/provider-physical-network.md

```diff
@@ -0,0 +1,116 @@
+
+### Attach VM to the external network
+
+Finally add a new NIC to a VM and select a profile of an external network. L2 connectivity should be provided to the VM. In case there is DHCP server running, VM should obtain an IP.
+
+## Caveat
+
+User needs to make sure that physical network used by external network is available on the host used by VM. There is currently now way for Engine how to enforce it. Easiest way how to accomplish this is to set physical network as required for the cluster.
+
+## Implementation
+
+### OpenStack Neutron API
+
+OpenStack Neutron API used by Engine to control external providers already provides means to specify physical network – attributes `provider:network_type`, `provider:physical_network` and `provider:segmentation_id`. Network types we use are `flat` and `vlan`. Physical network is matched by VDSM network name. Segmentation ID is used with VLAN networks and matches VLAN ID defined for selected network.
+
+### oVirt Provider OVN
+
+Provider reads attributes described above. If physical network is defined, provider implicitly creates `localnet` port on given network and attaches it to passed physical network. In case segmentation ID was passed, it uses it as a VLAN tag on the port.
+
+### VDSM
+
+Localnet port is attached to an abstract network name. This name is mapped to an OVS bridge name on each host via OVS DB attribute `external-ids:ovn-bridge-mappings`. This attribute is configured after each `setupNetworks` command and also during upgrade (after reboot or upgrade of vdsm package).
+
+### Engine
+
+`ProviderNetwork` object in Engine is extended with a link to its physical network (`Network`). In database, `provider_physical_network_id` is added as a new column to `network` table.
+
+Physical network is used to find VDSM network name and VLAN ID, these values are then passed to network provider.
+
+Engine also covers validation: physical network is on the same Data Center as the external network, custom values (physical network and VLAN ID) are not specified when provider physical network is set.
+
+### Engine UI
+
+Without this feature, there is already a way how to set physical network name and VLAN ID for an external network from `New Network` dialog. Physical network has its separate field, VLAN is obtained from shared `Network Attributes` section. However, this way requires user to know VDSM name of the network and also manually copy VLAN ID of the network.
+
+For better user experience, this feature exposes an option to select desired provider physical network from Data Center networks to UI.
+
+![add an external network connected to a physical network dialog](/images/features/network/provider-physical-network_new-network-dialog.png)
+
+To expose this feature, external OVN network is created implicitly when user adds a new VM network on OVS cluster. When selecting a network profile for a VM NIC, we list only external networks, not native oVirt networks that triggered their creation.
```

Comment:
To expose this feature, an external OVN network is created implicitly when the user adds a new VM network on an OVS cluster. When selecting a network profile for a VM NIC, we list only external networks, not native oVirt networks that triggered their creation.